frontend/src/components/common/Pagination.tsx

```diff
@@ -0,0 +1,134 @@
+  getLogRecords: (page_number: number) => Promise<void>;
+};
+
+const RESULTS_PER_PAGE_OPTIONS = [10, 25, 50];
+
+const Pagination = ({
+  numRecords,
+  resultsPerPage,
+  pageNum,
+  setResultsPerPage,
+  getLogRecords,
+}: Props): React.ReactElement => {
+  const numPages = Math.ceil(numRecords / resultsPerPage);
+
+  const handleNumberInputChange = (
+    newPageNumString: string,
+    newPageNum: number,
+  ) => {
+    if (newPageNum !== pageNum && newPageNum >= 1 && newPageNum <= numPages) {
+      getLogRecords(newPageNum);
+    }
+  };
+
+  const getNumRecordsStr = () => {
+    return `${numRecords} ${numRecords === 1 ? "item" : "items"}`;
+  };
+
+  return (
+    <Box padding="12px 0px 33px">
+      <Flex justifyContent="space-between" alignItems="center">
+        <Box>
+          <Text color="#6D8788">{getNumRecordsStr()}</Text>
+        </Box>
+        <Box>
+          <Flex alignItems="center" justifyContent="space-evenly">
+            <IconButton
+              className="icon-button"
+              variant="ghost"
+              aria-label="Previous page"
+              icon={<ChevronLeftIcon boxSize={7} color="#6D8788" />}
+              disabled={pageNum <= 1}
+              onClick={() => (pageNum > 1 ? getLogRecords(pageNum - 1) : null)}
+            />
+            <Flex
+              alignItems="center"
+              justifyContent="space-evenly"
+              padding="0px 25px 0px"
+              width="210px"
+            >
+              <Text>Page</Text>
+              <NumberInput
+                maxW="60px"
+                value={pageNum}
+                size="sm"
+                onChange={handleNumberInputChange}
+              >
+                <NumberInputField fontWeight="bold" />
+              </NumberInput>
+              <Text>of {numPages}</Text>
+            </Flex>
+            <IconButton
+              className="icon-button"
+              variant="ghost"
+              aria-label="Next page"
+              disabled={pageNum >= numPages}
+              icon={<ChevronRightIcon boxSize={7} color="#6D8788" />}
+              onClick={() =>
+                pageNum < numPages ? getLogRecords(pageNum + 1) : null
+              }
+            />
+          </Flex>
+        </Box>
+        <Box>
+          <Flex alignItems="center" justifyContent="space-evenly">
+            <Text paddingRight="8px">Show</Text>
+            <Menu>
+              <MenuButton
+                as={Button}
+                variant="outline"
+                width="73px"
+                rightIcon={<ChevronDownIcon color="#6D8788" />}
+              >
+                {resultsPerPage}
+              </MenuButton>
+              <MenuList>
+                {RESULTS_PER_PAGE_OPTIONS.map((opt, ind) => (
+                  <MenuItem
+                    key={ind}
+                    onClick={() => {
+                      // TO-DO - Confirm that resetting to page 1 is what's ideal
```

Comment:
so idk what the ideal behaviour is 😅 we can do two things:
- reset to page 1, ez
- figure out which page the user should be directed to. For example, page 4 with 25 results will display the same logs from page 2 with 50 results. 

Either is fine with me, but we should pick the solution that is more user-friendly

Reply:
noticing a bug here, steps to re-create:
1. Select "Show 10"
2. Go to page 5
3. Select "Show 25"
4. The input is outlined red, and says its on page 5 of 2. 
5. Clicking into the input and clicking out will navigate to page 2.

My guess is that it resets to page 1, but it doesn't update the actual page number. 
![image](https://github.com/uwblueprint/supportive-housing/assets/57375371/2ee1e556-218c-4978-a420-461e680a38ec)

Reply:
I'm also getting this error after recreating these steps!

Reply:
fixed 🙏